lib/posix.lua.in

```diff
@@ -519,5 +519,27 @@ if _DEBUG ~= false then
   end
 end
 
+local function prepare_args (pat_mark)
+  local pat, mark = "*", 0
+  if type (pat_mark) == "table" then
+    if type (pat_mark.pattern) == "string" then
+      pat = pat_mark.pattern
+    end
+    if pat_mark.MARK then
+      mark = M.GLOB_MARK
+    end
+  elseif type (pat_mark) == "string" then
+    pat = pat_mark
+  end
+
+  return pat, mark
+end
+
+local cglob = M.glob
+
+M.glob = function (...)
+  local pat, mark = prepare_args(...)
+  return cglob(pat, mark)
+end
 
```

Comment:
I tried to do that, but still fail the specs locally. This is as close as I can get:

    - posix file system glob with bad arguments diagnoses argument #1 type not table, string or nil:
      FAILED expectation 1: expecting an error containing any of:
      | bad argument #1 to '?' (table, string or nil expected, got boolean)
      or:
      | bad argument #1 to 'fn' (table, string or nil expected, got boolean)
      but got:
      | ./lib/posix/init.lua:541: bad argument #1 to 'glob' (table or string expected, got boolean)
    
    - posix file system glob with bad arguments diagnoses more than maximum of 1 arguments:
      FAILED expectation 1: expecting an error containing any of:
      | bad argument #2 to 'fn' (no more than 1 argument expected, got 2)
      or:
      | bad argument #2 to '?' (no more than 1 argument expected, got 2)
      but got:
      | ./lib/posix/init.lua:543: bad argument #2 to 'glob' (no more than 1 argument expected, got 2)

Here's the (still intermediate) code where I was trying to figure out how to satisfy `badargsdiagnose`. https://gist.github.com/476d562a266a934ea1a77c1014367743